src/translators/sql_translator.py

```diff
@@ -455,8 +455,175 @@ def is_text(attr_type):
+                    elif attr_t == NGSI_TEXT:
+                        mapped_value = self._ngsi_text_to_db(e[attr])
+                    elif attr_t == NGSI_DATETIME or attr_t == NGSI_ISO8601:
+                        mapped_value = self._ngsi_datetime_to_db(e[attr])
+                    elif attr_t == "Boolean":
+                        mapped_value = self._ngsi_boolean_to_db(e[attr])
+                    elif attr_t == "Number":
+                        mapped_value = self._ngsi_number_to_db(e[attr])
+                    elif attr_t == "Integer":
+                        mapped_value = self._ngsi_integer_to_db(e[attr])
+                    elif attr_t == 'Relationship':
+                        mapped_value = self._ngsi_ld_relationship_to_db(
+                            e[attr])
+                    elif self._is_ngsi_array(e[attr], attr_t):
+                        mapped_value = self._ngsi_array_to_db(e[attr])
+                    elif self._is_ngsi_object(e[attr], attr_t):
+                        mapped_value = self._ngsi_structured_to_db(e[attr])
+                    else:
+                        mapped_value = self._ngsi_default_to_db(e[attr])
+
+                    values.append(mapped_value)
+                except KeyError:
+                    # this entity update does not have a value for the column
+                    # so use None which will be inserted as NULL to the db.
+                    values.append(None)
+                except ValueError:
+                    # this value cannot be cast to column type
+                    # so use None which will be inserted as NULL to the db.
+                    values.append(None)
+        return values
+
+    @staticmethod
+    def _is_ngsi_array(attr, attr_t):
+        return (attr_t == NGSI_STRUCTURED_VALUE
+                and 'value' in attr and isinstance(attr['value'], list)) or attr_t == "Array"
+
+    @staticmethod
+    def _is_ngsi_object(attr, attr_t):
+        return attr_t == NGSI_STRUCTURED_VALUE or ('value' in attr and isinstance(attr['value'], dict))
+
+    @staticmethod
+    def _is_ngsi_ld_datetime_property(attr):
+        if 'type' in attr and attr[
+            'type'] == 'Property' and 'value' in attr and isinstance(
+                attr['value'], dict) \
+                and '@type' in attr['value'] and attr['value'][
+            '@type'] == 'DateTime':
+            return True
+        return False
+
+    @staticmethod
+    def _ngsi_geojson_to_db(attr):
         raise NotImplementedError
 
+    @staticmethod
+    def _ngsi_number_to_db(attr):
+        try:
+            if isinstance(attr['value'], bool):
+                return None
+            elif isinstance(attr['value'], float):
+                return attr['value']
+            elif attr['value'] is not None:
+                return float(attr['value'])
+        except ValueError:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
```

Comment:
Is this going to work in all cases? what if `attr` has no `type` attribute in it?